### PR TITLE
Migrate Slide Panel to CSS Modules.

### LINKED ISSUE #565: Convert Slide Panel component to CSS modules
# Issue Description
To better align with modern CSS practices and modern CSS theming capabilities, we want to update this component to use CSS modules and if applicable, CSS custom properties.

## Issue Type

- [ ] New Feature
- [x] Enhancement
- [ ] Bug
- [ ] Other

## Expected Behavior
Component styles are converted to use CSS modules. If applicable, convert SCSS variables to CSS custom properties.

## Current Behavior
Component styles use manual name spaced classes and SCSS variables.

### PR DESCRIPTION
**Pull request by emilyrohrbough**
### Summary
Closes #565.

@cerner/terra

### CODE REVIEW

File: packages/terra-slide-panel/src/SlidePanel.scss [hunk 1]
```diff
@@ -1,121 +1,124 @@
...
-  overflow-y: auto;
+/* stylelint-disable selector-class-pattern */
+:local {
+  .slide-panel {
+    overflow: hidden;
+    position: relative;
+  }
 
-  transition: transform 0.15s ease;
+  .main-panel,
```
Comment (tbiethman) [inline packages/terra-slide-panel/src/SlidePanel.scss]: Can we call this class '.main' instead?

Comment (emilyrohrbough) [inline packages/terra-slide-panel/src/SlidePanel.scss]: No problem. I changed it to .main-panel because it was initially SlidePanel-main

Comment (tbiethman) [inline packages/terra-slide-panel/src/SlidePanel.scss]: Thanks for updating that.

**Comment by PaulSolDev** [comment]
Functional review complete.  No issues identified.

**Pull request merged**